pkg/gateway/model_build_targetgroup.go

```diff
@@ -372,6 +391,23 @@ func (t *latticeServiceModelBuildTask) buildTargetGroupSpec(ctx context.Context,
 
 	tgName := latticestore.TargetGroupName(string(httpBackendRef.Name()), namespace)
 
+	tgp, err := getAttachedTargetGroupPolicy(ctx, client, string(httpBackendRef.Name()), namespace)
+	if err != nil {
+		return latticemodel.TargetGroupSpec{}, err
+	}
+	protocol := "HTTP"
+	protocolVersion := vpclattice.TargetGroupProtocolVersionHttp1
+	healthCheckConfig := &vpclattice.HealthCheckConfig{}
+	if tgp != nil {
+		if tgp.Spec.Protocol != nil {
+			protocol = *tgp.Spec.Protocol
+		}
+		if tgp.Spec.ProtocolVersion != nil {
+			protocolVersion = *tgp.Spec.ProtocolVersion
```

Comment:
do we need to do any translation on these for the API, like ```ToUpper()``` or anything?

Reply:
I can double check but I don't think so. we already have validations on YAML file.